packages/vscode-extension/src/utilities/packageManager.ts

```diff
@@ -32,15 +31,91 @@ export async function resolvePackageManager(): Promise<PackageManagerName> {
     } else if (isNpm) {
       return "npm";
     }
+    try {
+      const packageManager = require(path.join(workspacePath, "package.json")).packageManager;
+
+      if (packageManager) {
+        const regex = /^([a-zA-Z]+)@/;
+        const match = packageManager.match(regex);
+        return match ? match[1] : "npm";
+      }
+    } catch (e) {
+      // there might be a problem while reading package.json in which case we default to npm
+    }
 
     return "npm";
   });
 }
 
 export function isPackageManagerAvailable(manager: PackageManagerName): boolean {
   try {
-    exec(`${manager} --version`);
+    command(`${manager} --version`);
     return true;
   } catch {}
   return false;
 }
+
+async function isNpmModulesInstalled(): Promise<boolean> {
+  const workspacePath = getAppRootFolder();
+  try {
+    const { stdout, stderr } = await command("npm ls --json", {
+      cwd: workspacePath,
+      quiet: true,
+    });
+    const parsedJson = JSON.parse(stdout);
+    return parsedJson.problems ? false : true;
+  } catch (e) {
+    return false;
+  }
+}
+
+async function isYarnModulesInstalled(): Promise<boolean> {
+  const workspacePath = getAppRootFolder();
+  try {
+    // because "yarn check" was removed from yarnv2 we use npm's method for checking dependencies
+    // npm is taking into consideration yarn.lock since version 7, which means
+    // that if the users shell is using the older one this function may produce false in unexpected ways but even then
+    // we'll just run "yarn install" every time which is exactly what we would need to do without isYarnInstalled.
+    // https://docs.npmjs.com/cli/v7/commands/npm-install
+    const { stdout, stderr } = await command("npm ls --json", {
+      cwd: workspacePath,
+      quiet: true,
+    });
+    const parsedJson = JSON.parse(stdout);
+
+    // because npm marks packages installed with yarn as "extraneous" we need to check if there are any other problems.
+    if (parsedJson.problems) {
+      parsedJson.problems.forEach((element: string) => {
+        if (!element.startsWith("extraneous")) {
+          return false;
+        }
+      });
+    }
+    return true;
```

Comment:
If we do inverse, we can exit early. Also, `return` in `forEach` doesn't work as intended.


```ts
for(const problem of parsedJson.problems ?? []) {
  if(element.startsWith("extraneous")) { return true; }
}
```